core/src/main/java/org/infinispan/metrics/impl/MetricsCollector.java

```diff
@@ -109,30 +109,40 @@ private static String generateRandomName() {
    }
 
    public Set<Object> registerMetrics(Object instance, Collection<MBeanMetadata.AttributeMetadata> attributes, String namePrefix, String cacheName) {
-      Set<Object> metricIds = new HashSet<>();
+      return doRegisterMetrics(instance, attributes, namePrefix, cacheName, nodeTag);
+   }
+
+   public Set<Object> registerMetrics(Object instance, Collection<MBeanMetadata.AttributeMetadata> attributes, String namePrefix, String cacheName, String nodeName) {
+      return doRegisterMetrics(instance, attributes, namePrefix, cacheName, nodeName == null ? null : new Tag(NODE_TAG_NAME, nodeName));
```

Comment:
```suggestion
      return doRegisterMetrics(instance, attributes, namePrefix, cacheName, nodeName == null ? null : nodeTag);
```

Reply:
Ah okay I thought it was the same, ignore then.

Reply:
Actually, I think this is correct the way I proposed. You have to use both this change and the other together.